### PR TITLE
refactor legenda to styled component and fix layout bug

### PR DESCRIPTION
**Pull request by VWSCoronaDashboard5**
## Summary

Due to the large legend labels, the legend `boxes` would be pushed away from each other. I refactored the `ChoroplethLegenda` to `styled-components` and implemented a variable `width` based on the media query:
```
const LegendaItemBox = styled(Box)(
  css({
    width: ['50px', null, '60px'],
    height: '10px',
  })
);
```

### Governance

- [x] Documentation is added
- [x] Test cases are added or updated
- [x] I've read the contributing document https://github.com/minvws/.github/blob/master/CONTRIBUTING.md
- [x] I've read and understand the Code of Conduct https://github.com/minvws/.github/blob/master/CODE_OF_CONDUCT.md
- [x] I understand that any contributions or suggestions I made may make it into the actual code. I've read the License
      https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/LICENSE.txt
      and the contributor license agreement https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/CLA.md

### CODE REVIEW

File: src/components-styled/choropleth-legenda.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    <Box>
+      <h4>{title}</h4>
+      <Ul aria-label="legend">
+        {items.map((item, index) => (
+          <Li key={item.color}>
+            <LegendaItemBox
+              borderLeft={index === 0 ? '1px solid lightgrey' : null}
+              borderRight={
+                index === items.length - 1 ? '1px solid lightgrey' : null
+              }
```
Comment (VWSCoronaDashboard1) [inline src/components-styled/choropleth-legenda.tsx:58]: Can this be solved in css? `&:first-child` and `&:last-child`?

**Review by VWSCoronaDashboard1 — APPROVED** [review]

File: src/components-styled/choropleth-legenda.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  const { items, title } = props;
+
+  return (
+    <Box>
+      <h4>{title}</h4>
+      <Ul aria-label="legend">
+        {items.map((item, index) => (
+          <Li key={item.color}>
+            <LegendaItemBox
+              borderLeft={index === 0 ? '1px solid lightgrey' : null}
```
Comment (VWSCoronaDashboard4) [inline src/components-styled/choropleth-legenda.tsx:55]: You can use
```
"&:first-child" : {
   borderLeft: ...
}
```
in the LegendaItemBox, same for `"&:last-child"`. See the use of `"&:focus"` in other components.

File: src/components-styled/choropleth-legenda.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  color: string;
+  label: string;
+}
+
+export type TProps = {
+  title: string;
+  items: ILegendaItem[];
+};
+
+const Ul = styled.ul(
```
Comment (VWSCoronaDashboard4) [inline src/components-styled/choropleth-legenda.tsx:15]: Maybe name this one `List` and the `Li` `Item` instead? Not sure what styled system convention is.

File: src/components-styled/choropleth-legenda.tsx (hunk 1 above)
Comment (VWSCoronaDashboard5) [reply src/components-styled/choropleth-legenda.tsx:58]: yes. I was looking for this but couldn't find it in the docs. thanks!

File: src/components-styled/choropleth-legenda.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import { Box } from './base';
+import styled from 'styled-components';
+import { css } from '@styled-system/css';
+
+export interface ILegendaItem {
```
Comment (VWSCoronaDashboard8) [inline src/components-styled/choropleth-legenda.tsx:5]:
```suggestion
export interface LegendaItem {
```

File: src/components-styled/choropleth-legenda.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  color: string;
+  label: string;
+}
+
+export type TProps = {
```
Comment (VWSCoronaDashboard8) [inline src/components-styled/choropleth-legenda.tsx:10]:
```suggestion
export type ChoroplethLegendaProps = {
```

File: src/components-styled/choropleth-legenda.tsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  items: ILegendaItem[];
+};
+
+const Ul = styled.ul(
+  css({
+    marginTop: 0,
+    paddingLeft: 0,
+    listStyle: 'none',
+    display: 'inline-flex',
+    flexDirection: 'row',
```
Comment (VWSCoronaDashboard8) [inline src/components-styled/choropleth-legenda.tsx:21]: nit; row is default so this would be redundant I think

**Review by VWSCoronaDashboard8 — APPROVED** [review]

File: src/components-styled/choropleth-legenda.tsx [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+    fontSize: [0, null, 1],
+    '&:first-child div:first-child': {
+      borderLeft: '1px solid lightgrey',
+    },
+    '&:last-child div:first-child': {
+      borderRight: '1px solid lightgrey',
+    },
```
Comment (VWSCoronaDashboard5) [inline src/components-styled/choropleth-legenda.tsx:35]: This must look cumbersome to you. When I assign the styles to `LegendaItemBox` it wouldn't have a context of `first-child`:

```
const LegendaItemBox = styled(Box)(
  css({
    width: ['50px', null, '60px'],
    height: '10px',
    flexGrow: 0,
    flexShrink: 0,
    borderBottom: '1px solid lightgrey',
    borderTop: '1px solid lightgrey',
    '&:first-child': {
      borderLeft: '1px solid lightgrey'
     }
  })
);
```
this does not work.

So I assigned it using its parent component.

**Pull request merged**